src/ert/run_models/everest_run_model.py

```diff
@@ -381,6 +408,54 @@ def _create_optimizer(self) -> BasicOptimizer:
             ),
         )
 
+        def _forward_ropt_event(everest_event: OptimizerEvent) -> None:
+            has_results = bool(everest_event.data and everest_event.data.get("results"))
+
+            # The batch these results pertain to
+            # If the event has results, they usually pertain to the
+            # batch before self._batch_id, i.e., self._batch_id - 1
+            batch_id = (
+                everest_event.data["results"][0].batch_id
+                if has_results
+                else self._batch_id
+            )
+
+            data_json = {}
+            if has_results:
+                results = everest_event.data["results"]
+                exit_code = everest_event.data["exit_code"]
+
+                result_types = [
+                    "FunctionResults"
+                    if isinstance(r, FunctionResults)
+                    else "GradientResults"
+                    for r in results
+                ]
+
+                data_json = {
+                    "exit_code": exit_code,
+                    "results": {},
+                }
+
+                if any(isinstance(r, GradientResults) for r in result_types):
+                    data_json["kind"] = "gradient_evaluation"
+
+                if any(isinstance(r, FunctionResults) for r in result_types):
+                    data_json["kind"] = "function_evaluation"
```

Comment:
data_json["kind"] get overwritten is this on purpose? 

Also can you please consider renaming `kind` to `type`. If you are worried about name clashes with the global `type` in the future if we make the `data_json` dict into a dataclass, you can also consider `result_type`.